tokio/src/sync/broadcast.rs

```diff
@@ -192,8 +195,8 @@ pub struct Receiver<T> {
     /// Next position to read from
     next: u64,
 
-    /// Waiter state
-    wait: Arc<WaitNode>,
+    /// Used to support the deprecated `poll_recv` fn
+    waiter: Option<Pin<Box<UnsafeCell<Waiter>>>>,
```

Comment:
Is this boxed so that `Receiver` can stay `Unpin`? Or in order to to reduce the struct size if the functionality is not used?

Reply:
For it to stay `Unpin`, as I understand it. The `poll_recv` method takes `&mut self`.

Reply:
@Matthias247 Why not both? Primarily because `Receiver` is `Unpin` (a requirement).